oap-server/server-storage-plugin/storage-influxdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/influxdb/query/AlarmQuery.java

```diff
@@ -64,7 +64,7 @@ public Alarms getAlarm(Integer scopeId, String keyword, int limit, int from, lon
                        .and(lte(InfluxClient.TIME, InfluxClient.timeInterval(endTB)));
         }
         if (!Strings.isNullOrEmpty(keyword)) {
-            recallQuery.and(regex(AlarmRecord.ALARM_MESSAGE, keyword));
+            recallQuery.and(contains(AlarmRecord.ALARM_MESSAGE, keyword.replaceAll("/", "\\/")));
```

Comment:
```suggestion
            recallQuery.and(contains(AlarmRecord.ALARM_MESSAGE, keyword.replaceAll("/", "\\\\/")));
```

Reply:
Why four \? How many do we need?

@kezhenxu94 we may need an e2e query for alarm in the future.

Reply:
> Why four ? How many do we need?
> 

`"\\"` => `\`, `\` in regexp pattern is for escaping, so `\\\\` in string literal => `"\\(escape in string literal)\\(escape in string literal)"` => `\\` (escape `\` in regexp)

Reply:
> @kezhenxu94 we may need an e2e query for alarm in the future.

please create an issue to track in case we all forget

Reply:
> > Why four ? How many do we need?
> 
> `"\\"` => `\`, `\` in regexp pattern is for escaping, so `\\\\` in string literal => `"\\(escape in string literal)\\(escape in string literal)"` => `\\` (escape `\` in regexp)

@wu-sheng @dmsolr if you're not sure, please refer to the JavaDoc of `java.util.regex.Matcher#replaceAll` or simply run the following test:

```java
    public static void main(String[] args) {
        System.out.println("/abc/".replaceAll("/", "\\/"));
        // still produce /abc/ , not \/abc\/
        System.out.println("/abc/".replaceAll("/", "\\\\/"));
        // produce \/abc\/ , which is what we want,
    }
```

Reply:
Got it